src/read.c

```diff
@@ -4401,25 +4411,14 @@ static avifResult avifMetaFindAlphaItem(avifMeta * meta,
         }
     }
     assert(alphaItemCount == colorItemCount);
-    // Find an unused ID.
+    // Figure out the last used itemID.
     avifResult result;
-    if (meta->items.count >= UINT32_MAX - 1) {
-        // In the improbable case where all IDs are used.
+    const uint32_t lastID = meta->items.count ? meta->items.item[meta->items.count - 1]->id : 0;
```

Comment:
At this point, we know `meta->items.count` is nonzero. Reason: The `if` statement at line 4376 ensures `colorItemCount` is nonzero. The assertion at line 4413 implies `alphaItemCount` is nonzero. If `alphaItemCount` is nonzero, then we must have entered the for loop at line 4384. So `meta->items.count` is nonzero.

So this can be simplified to:

```
    const uint32_t lastID = meta->items.item[meta->items.count - 1]->id;
```

Reply:
Good catch, done.